### DISCUSSION TITLE
Establishing pregnant woman population simulation need your help

### DISCUSSION THREAD

**Discussion by D-SYD** [Open-Systems-Pharmacology/Forum#975] [How-Tos]
Hello, everyone:
	I’m a beginner learning PBPK simulation.
	Thanks for your attention.
	I want to establish a simulation of pregnant population with PK-SIM and MOBI (v9.1), thus I have followed the tutorial but failed, and found something strange (show later) and have some question that need your help (①②③④). 

①	I want to know does the tutorial still apply to the current version of PK-SIM and MOBI? IF not，what processes should I go through to build a simulation of pregnant population？
	
Tutorial: Dallmann A, Solodenko J, Ince I, Eissing T. Applied Concepts in PBPK Modeling: How to Extend an Open Systems Pharmacology Model to the Special Population of Pregnant Women. CPT Pharmacometrics Syst Pharmacol. 2018 Jul;7(7):419-431. doi: 10.1002/psp4.12300. Epub 2018 Apr 20. PMID: 29569837; PMCID: PMC6063743.

I have tried many times following the tutorial but still failed, so I find the tutorial project file on the official forum (really thanks for Dallmann’s sharing) (https://github.com/Open-Systems-Pharmacology/Pregnancy-Models/tree/master/Models/Metronidazole)

Here are some of the strange situations and questions：
1.	I open the file （Dallmann’s sharing）, double-click "Non-pregnant_Houghton1979" simulation, without any other operation, then I can see the analysis figure.
 
<img width="473" alt="1" src="https://user-images.githubusercontent.com/79398922/109778523-e1eb2500-7c3f-11eb-91ef-81febf397529.png">

But when I click “Configure” and click ”OK” directly, then click “Run”, the figure change!
 
<img width="480" alt="2" src="https://user-images.githubusercontent.com/79398922/109778548-e9aac980-7c3f-11eb-9d44-945de360de33.png">

Thus, I spend several hours trying to find out the cause of figure change. Then I find some parameters change automatically when I click “OK” after “Configure” without any other operation.

Before
 
<img width="779" alt="3" src="https://user-images.githubusercontent.com/79398922/109778562-ee6f7d80-7c3f-11eb-9c6e-496ff3966978.png">

After
 
<img width="769" alt="4" src="https://user-images.githubusercontent.com/79398922/109778573-f29b9b00-7c3f-11eb-9dee-5b7bd4694aad.png">

2．	The “Spatial Structures” file of pregnant woman provided in tutorial (Open-Systems-Pharmacology: Pregnancy-Models/Building Blocks. <https://github.com/Open-Systems-Pharmacology/Pregnancy-Models/tree/master/BuildingBlocks>. Accessed 17 March 2018) is different with the file provided by Dallmann’s sharing.
The differences can be found in “Parameters”→“Organism”→the nine additional pregnancy-specific compartments “Parameters” like “Weight”,”Organ volumes” …

3．The “Parameter start values” in Dallmann’s file contains the nine additional pregnancy-specific compartments, So should I do the same as what tutorial said as “molecule start values”?

Did I do something wrong or set up the software wrong?

②	I don’t know how to deal with situation 1 (parameters change automatically in PK-SIM).
③	Can I use the “Spatial Structures” from Dallmann’s sharing file in my simulation by save as PKML?
④	What should I do with “Parameter Star values” setting?

Thanks a lot
Best wishes

**Comment by StephanSchaller** [comment]
@AndreDlm ;-)

**Comment by msevestre** [comment]
Hi @D-SYD 

I think there is a lot of stuff going on here. Some may be bugs so we'll have to dig deeper.

>I have tried many times following the tutorial but still failed,

Please review https://github.com/Open-Systems-Pharmacology/Suite/blob/develop/CONTRIBUTING.md#reporting-a-bug for details on how to report a bug. Specifically, we need to understand **what** is not working...otherwise, we can't help

>But when I click “Configure” and click ”OK” directly, then click “Run”, the figure change!

When you click configure, a NEW simulation is created using the building blocks. So it could be that whatever simulation you were using does not match the new structure.
A good way to see what has changed would be to CLONE the simulation, then perform a DIFF of the two simulations (by using the comparison tool). That should give you a good first guess as to what has changed (or us if this is a bug)

You can also add your project here so that we can try to understand what's going on (you'll have to zip it first)

For the rest, I'll let the pregnancy model expert respond

**Comment by D-SYD** [comment]
Thanks a lot @msevestre 

Thank you for helping me this beginner, I will review the link first.

Thanks again

**Comment by AndreDlm** [comment]
Hi @D-SYD 

I'm not sure I understand the issue correctly, but just few thoughts on some of your comments:

> 2． The “Spatial Structures” file of pregnant woman provided in tutorial (Open-Systems-Pharmacology: Pregnancy-Models/Building Blocks. https://github.com/Open-Systems-Pharmacology/Pregnancy-Models/tree/master/BuildingBlocks. Accessed 17 March 2018) is different with the file provided by Dallmann’s sharing.
> The differences can be found in “Parameters”→“Organism”→the nine additional pregnancy-specific compartments “Parameters” like “Weight”,”Organ volumes” …

That is what I would expect. As described in the [tutorial ](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12300), the spatial structure `pkml` file only contains the spatial structure, but not the correct parameterization, i.e. the values assigned to different parameters are **not** relevant. They are only placeholders and will be overwritten by the correct pregnancy values when importing the `pkml` (of the simulation) into PK-Sim and coupling it with the pregnancy population. 

> 3．The “Parameter start values” in Dallmann’s file contains the nine additional pregnancy-specific compartments, So should I do the same as what tutorial said as “molecule start values”?

No, please follow the [tutorial ](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12300) if you want to run a pregnancy population simulation. Otherwise you may overwrite e.g. enzyme expression values in the PSV building block...

I hope this helps
Best regards
André

**Comment by D-SYD** [comment]
@msevestre 
Sorry for the delayed communication with you

**Description of problem:**

I am learning to establish a pregnant women population simulation by following the tutorial. But I find out that, in step one, establish a non-pregnant woman individual simulation, even though my settings (details are described below) are consistent with the Dallmann’s, I can’t reproduce the result. 

Then I find out the simulation parameters of individuals fluid recirculation flow rat will change automatically when simulation run what may cause the different result. (At the end of this post)

**Reproduction Steps:**
PS: I use the PK-Sim file provided by Dallmann on the official forum (Link: https://github.com/Open-Systems-Pharmacology/Pregnancy-Models/tree/master/Models/Metronidazole)

----------------------**-my simulation and setting details-**----------------
1.	Open the file “Metronidazole_Pregnancy.pksim5”
2.	Click “Simulation” to establish a new simulation
3.	Define the new simulation’s name as TEST, then change nothing and click “Next” 
4.	Change “Partition coefficients” as “Rodgers and Rowland”, then click “Next”
5.	Change “Renal Clearances” as “CLren” and change “Glomerular Filtration” as “None”, then click “Next”
![image](https://user-images.githubusercontent.com/79398922/110206987-0a7f5300-7ebc-11eb-8a60-8c5ae643132e.png)
6.	Without any change in “Administration” and “Events” page, click “Next” directly respectively
7.	Without any change in “Observers” page and click “OK” directly
8.	Click “Run” 
9.	Select the curves that will be generated by the simulation, like “Kidney→Urine/Metronidazole/Fraction excreted to urine”, “Liver→Intracellular/Metronidazole-CYP3A4-Pearce 2013 Metabolite/Fraction of dose-Metronidazole” and “Liver→Intracellular/Metronidazole-CYP2A6-Pearce 2013 Metabolite/Fraction of dose-Metronidazole”, then click “OK”
![image](https://user-images.githubusercontent.com/79398922/110206992-0f440700-7ebc-11eb-8e3d-d563c0df6e74.png)
10.	Draw the observed data (DoseFraction_CYP2A6, DoseFraction_CYP3A4, DoseFraction_UnchangedUrine, ObsData_NP_Houghton1979_IV) to the simulation “TEST” time profile analysis figure
11.	Right-click “TEST” on the left side of the screen, and click “Start parameter Identification…”, then create a new parameter identification page named “Parameter Identification 2” (Dallmann’s named “Parameter Identification 1”)
12.	Click “Add output” on the right side of the screen, select the output as “TEST-Metronidazole-CYP2A6-Pearce 2013 Metabolite-Liver-Intracellular-Fraction of dose-Metronidazole” and the Observed data as “Dose fraction CYP2A6”
13.	Then change the values of “Weight” as Dallmann’s file’s ( DoseFraction_CYP2A6, CYP3A4 and UnchangeUrien are 2 )
14.	Then go to “Parameters” page, add “Kidny→Specific clearance/Metronidazole”, “Reaction→Specific clearance/Metronidazole”, “Reaction→kcat/ Metronidazole -CYP3A4-Pearce 2013” and “Reaction→kcat/ Metronidazole -CYP2A6-Pearce 2013” to the “Identification Parameters” by click the “add＞＞” buttom
15.	Then set “Start Value”, “Min. Value”, “Max. Value” and “Scaling” same as Dallmann’s file’s 
![image](https://user-images.githubusercontent.com/79398922/110207005-1703ab80-7ebc-11eb-9fd6-111f655373f5.png)
16.	Then go to the “configuration” page of “Parameter Identification”, change the “Algorithm” to “Monte - Carlo” and tick the “Calculate Sensitivity”
17.	Click “Run” and obtain the fitted result



**Expected behavior:**
The “CYP3A4 kcat”, “CYP2A6 kcat”, “Specific hepatic clearance” and “Specific renal clearance” values fitted by myself should be same as Dallmann’s.

**Observed behavior:**

The values are different with mine and Dallmann’s.

 
![image](https://user-images.githubusercontent.com/79398922/110207024-2551c780-7ebc-11eb-81bc-d664b240dc35.png)
Dallmann’s values

 
![image](https://user-images.githubusercontent.com/79398922/110207031-297de500-7ebc-11eb-8dd9-5acbd18d7d3f.png)
My values


----------**Find out the parameters changing automatically**---------
1.	Double-click the simulation named “Non-pregnant_Houghton1979” (individual), open the “Parameters” page, select the “Advanced” model, click ”Physiology”, “Flow rates” and “Fluid recirculation flow rate” in order.
2.	Screen shot the values like following 
![image](https://user-images.githubusercontent.com/79398922/110207035-2f73c600-7ebc-11eb-90c6-a6a696f83a5a.png)
3.	The values above are same as the values of individual parameters (Non-Pregnant_Houghton1979), means that, the simulation named “Non-Pregnant_Houghton1979” calculating with the individual block named “Non-Pregnant_Houghton1979”
4.	Select the simulation named “Non-Pregnant_Houghton1979” by double-click, then click “configure” button and click “OK” without any other operation.
5.	Then I found an exclamation mark on the “Analysis” icon of the “Non-Pregnant_Houghton1979” simulation
 ![image](https://user-images.githubusercontent.com/79398922/110207037-34387a00-7ebc-11eb-94e9-b8f2c588a3b7.png)
6.	So I click “Run”, then the red solid line change (lower)
 ![image](https://user-images.githubusercontent.com/79398922/110207037-34387a00-7ebc-11eb-94e9-b8f2c588a3b7.png)
7.	Now I repeat the step 1 to check and screen shoot the “Fluid recirculation flow rate” values. The values is change.
 ![image](https://user-images.githubusercontent.com/79398922/110207044-3bf81e80-7ebc-11eb-9099-a1f28f3d9ca4.png)

**Expected behavior:**
Because I don’t change any parameters or settings, nothing change with simulation “Non-Pregnant_Houghton1979” is expected.

**Observed behavior:**
The “Fluid recirculation flow rate” values change. Is that the cause of simulation result (red solid line) change?

**Comment by D-SYD** [comment]
[Uploading Metronidazole_Pregnancy after configure.zip…]()

**Comment by msevestre** [comment]
@D-SYD  wow!!. You just brought the bug reporting to the next level! Thank you so much for all your step by step instructions.
I'll review next week and let you know

**Comment by D-SYD** [comment]
@AndreDlm 
Sorry for the delayed communication with you.

I should have misunderstood something. I will reread the tutorial.
After reorganizing my questions, I will communicate with you.
Thanks for your attention and thanks for your help.

Thanks & Best regards

**Comment by D-SYD** [comment]
Hi @AndreDlm 

Sorry for the delayed communication with you.
I tried many times but still can't get the same results as the file. I think it must be my problem. Can you help me find where I am doing something wrong?
I will write my complete process from scratch below.

-----------------establish a non-pregnant woman simulation------------

1.	I establish a individual named “I2”, European, female, and use system default values for other individual parameters. Without any change about “anatomy & physiology”, then add CYP3A4 and CYP2A6 as metabolizing enzymes, both of them use “RT-PCT” expression data.

2.	I establish a compound named “C2”, with the same values of the basic physico-chemistry  parameters as “Metronidazole”(Dallmann’s file), then add CYP2A6 and CYP3A4 as the metabolizing enzymes in “ADME” page, both of them use the “In vitro metabolic rate in the presence of liver microsomes – Michaelis-Menten” process type, the values of Km are same as tutorial.
Then, add total hepatic clearance process with “Liver Plasma Clearance” process type, and click “OK” without any change.
And add renal clearance process with “Kidney plasma clearance” process type, and click “OK” without any change.

3.	Establish a simulation named S2 with I2, C2 and the administration protocol provided form Dallmann’s file named “IV_500mg”, with the “Rodgers and Rowland” partition coefficients calculation method.

4.	Run the S2 simulation and finish the individual simulation settings by adding the fraction excreted to urine curve, intracellular- C2- CYP3A4- metabolite- fraction of dose curve and intracellular- C2- CYP2A6- metabolite- fraction of dose curve.
 

5.	Draw the observed data “DoseFraction_CYP2A6”, “DoseFraction_CYP3A4”, “DoseFraction_UnchangedUrine” and “ObsData_NP_Houghton1979_IV”, provided by Dallmann’s file, to the S2 simulation time profile analysis.

6.	Click “Start Parameters Identification” by right click the “S2” in “Simulation” frame, rename as “P2”. Then, set every values same as Dallmann’s file including “output weight”, “start value of identification parameters”, “Max. value of identification parameters” and algorithm type.
The results shown below is different form Dallmann’s.
 
![1](https://user-images.githubusercontent.com/79398922/111021783-d18a3580-8409-11eb-8d54-a4f86c80ed78.png)


7.	For following the tutorial, I change the “Biochemical processes” and “compounds” values of simulation “S2” as tutorial adjusting for pregnant women.

----------------Work with MOBI------------------

1.	Right click the “S2” simulation and select “send to MOBI”.

2.	Load the spatial structures and passive transports as tutorial said.

3.	Double click the passive transport named “PregnantWoman” and select the ”Renal Clearances-renal plasma clearance”, then add the molecule “C2” in the “Include list”， then click “parameters”, find “Specific clearance” and change the value to 0.0222.

 
![2](https://user-images.githubusercontent.com/79398922/111021788-db139d80-8409-11eb-8147-e117329c34fa.png)
![3](https://user-images.githubusercontent.com/79398922/111021790-dc44ca80-8409-11eb-8bdc-db36063cecab.png)


4.	Create a new molecule start values building block named “TMP S2 MSV”, with the “S2” molecule building block and “PregnantWoman” spatial structure, then save as a PKLM file.

5.	Select the molecule start value named “S2”, select “merge” by right click and operate as tutorial said.

6.	***********Set the molecule start value named “S2”*************
 
![4](https://user-images.githubusercontent.com/79398922/111021799-e5ce3280-8409-11eb-8310-3bf309e25b41.png)
![5](https://user-images.githubusercontent.com/79398922/111021801-e6ff5f80-8409-11eb-9915-f3d5952d90e8.png)
![6](https://user-images.githubusercontent.com/79398922/111021803-e6ff5f80-8409-11eb-8b16-ae47e7ac7fc0.png)

7.	Establish a simulation named “S2-M2” with tutorial setting and save the “S2-M2 MOBI” as PKML format.

-------------------Back to PK-Sim----------------------

1.	Import population simulation by the file named “S2-M2 MOBI”.

2.	Create a new population named “S2-M2 pregnant population” by create a new individual named “S2-M2 pregnant individual”.

3.	Create a new individual named “S2-M2 pregnant individual” with adding the CYP2A6 and CYP3A4 as the metabolizing enzymes with RT-PCR expression type.

4.	Set the number of individuals to 2000 and the population age from 30.69 to 30.73, without other change.

5.	The population simulation is established named “S2-M2”. Create a time profile analysis as below setting.
 
![7](https://user-images.githubusercontent.com/79398922/111021826-fed6e380-8409-11eb-9798-40d06cb6578b.png)


6.	The result is different from Dallmann’s, especially the dose fraction excreted unchanged in urine and the dose fraction metabolized via CYP2A6/ CYP3A4.
![8](https://user-images.githubusercontent.com/79398922/111021828-039b9780-840a-11eb-8f93-4cb05d206579.png)
![9](https://user-images.githubusercontent.com/79398922/111021830-04342e00-840a-11eb-8e89-147542868876.png)

Dallmann’s result
![10](https://user-images.githubusercontent.com/79398922/111021835-09917880-840a-11eb-8af5-a60fc9040bc0.png)


I compare the simulation established by myself and Dallmann’s simulation. I find out the parameters “Anatomy” and “Physiology” is very different. 

I think there must be something wrong with my operation in MOBI. 
Really need your help.

How to upload files bigger than 10MB?
Best regards
SYD

**Comment by D7abc** [comment]


**Reply by AndreDlm** [reply]
Dear @D7abc, I am not sure I understood your question correctly, but here are a few general remarks that might be helpful: 

You need to create a pregnant woman model separately for each drug that you want to model according to this [tutorial](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12300). 

When you have created a pregnancy model for a given drug, you do not need to add new parameters; however, you need to manually change a few values of parameters that are altered in pregnancy and that are **not** implemented in the PK-Sim physiological and anatomical database (e.g., because they are located in the `compounds` building block and not in the `individual` building block). 
In most cases, these parameters include the fraction unbound in plasma (f<sub>up</sub>) as well as k<sub>cat</sub> values of enzymes/transporters with increased or decreased activity during pregnancy that are involved in clearance/distribution of the compound in your model. 

There's one exception, though: If you would like to use the updated `spatial structure` and `passive transports` building blocks for pregnancy that include the fetal  f<sub>up</sub> (in addition to the maternal  f<sub>up</sub>), you need to add that parameter and its value in MoBi. You can find further information [here](https://github.com/Open-Systems-Pharmacology/Pregnancy-Models#accounting-for-different-unbound-drug-fractions-in-maternal-and-fetal-compartments).

Once you have prepared your pregnancy `pkml` file for a given drug as described in the [tutorial](https://ascpt.onlinelibrary.wiley.com/doi/full/10.1002/psp4.12300), you can import it in PK-Sim as population simulation and combine it with several populations of different gestational age ranges (e.g., first, second, and third trimester). You will thus have 3 different population simulations in PK-Sim (one for each trimester) that you need to run separately. 

I hope this helps! 
If you have further questions, kindly open a new issue in the Forum. Thanks!

**Reply by D7abc** [reply]
Thank you very much for providing me with ideas. If there are any questions in the future, I will release them separately. Thank you again

**Reply by AndreDlm** [reply]
Sure, you're welcome!
